bpf/deserialization_to_bpf_map/deserialization_to_bpf_map.c

```diff
@@ -771,6 +777,56 @@ static int repeat_field_query(struct op_context *ctx, const ProtobufCFieldDescri
     return ret;
 }
 
+void deserial_free_elem_list(struct element_list_node *head)
+{
+    while (head != NULL) {
+        struct element_list_node *n = head;
+        deserial_free_elem(n->elem);
+        head = n->next;
+        free(n);
+    }
+}
+
+static void *create_struct_list(struct op_context *ctx, int *err)
+{
+    void *prev_key = NULL;
+    void *value;
+    struct element_list_node *elem_list_head = NULL;
+    struct element_list_node *curr_elem_list_node = NULL;
+
+    *err = 0;
+    ctx->key = calloc(1, ctx->curr_info->key_size);
+    while (!bpf_map_get_next_key(ctx->curr_fd, prev_key, ctx->key)) {
+        prev_key = ctx->key;
+
+        value = create_struct(ctx, err);
+        if (*err != 0) {
+            LOG_ERR("create_struct failed, err = %d\n", err);
+            deserial_free_elem_list(elem_list_head);
+            value = NULL;
+        }
+
+        if (value == NULL) {
+            continue;
+        }
+
+        struct element_list_node *new_node = (struct element_list_node *)calloc(1, sizeof(struct element_list_node));
+        if (!new_node) {
+            deserial_free_elem_list(elem_list_head);
+            return NULL;
+        }
+        new_node->elem = value;
```

Comment:
These are basic operations for updating list. It is recommended that you encapsulate them into independent functions or macros or call existing libraries.

Reply:
thanks, will modify it.